### PR TITLE
[common][core][format] Support using file index result for RowGroup filtering.

### PR DESCRIPTION
**Pull request by hang8929201**
### Purpose

Use file index result(bitmap) for more fine-grained IO clipping in the reader.

### Tests
This is a query in our real business scenario.
- no index:
image
- index for parquet block filtering:
image
- index for orc rowgroup filtering:
image

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/operation/RawFileSplitRead.java [hunk 1]
```diff
@@ -191,26 +192,38 @@ private RecordReader<InternalRow> createFileReader(
             BulkFormatMapping bulkFormatMapping,
             IOExceptionSupplier<DeletionVector> dvFactory)
             throws IOException {
+        FileIndexResult fileIndexResult = null;
         if (fileIndexReadEnabled) {
-            boolean skip =
-                    FileIndexSkipper.skip(
+            //            boolean skip =
```
[the remark this answers is not in the document]
Comment (hang8929201) [reply paimon-core/src/main/java/org/apache/paimon/operation/RawFileSplitRead.java]: done

**Review by JingsongLi — CHANGES_REQUESTED** [review]

**Comment by JingsongLi** [comment]
Left comments @hang8929201

File: paimon-format/src/main/java/org/apache/paimon/format/orc/OrcReaderFactory.java [hunk 2]
```diff
@@ -338,6 +344,14 @@ public static org.apache.orc.Reader createReader(
...
-        return OrcFile.createReader(hPath, readerOptions);
+        return new ReaderImpl(hPath, readerOptions) {
+            @Override
+            public RecordReader rows(Options options) throws IOException {
+                return new RecordReaderImpl(this, options, fileIndexResult);
+            }
+        };
     }
 }
+/*
```
Comment (JingsongLi) [inline paimon-format/src/main/java/org/apache/paimon/format/orc/OrcReaderFactory.java:355]: ???

**Review by JingsongLi — APPROVED** [review]
Looks good to me! Thanks @hang8929201

**Pull request merged**